### PR TITLE
Fix empty MarchRobot by constructing it via the initializer list

### PR DESCRIPTION
**Pull request by Ishadijcks**
Fix a bug where the hardware_interface would refer to an empty object instead of the one created

### CODE REVIEW

File: march_hardware/src/MarchRobot.cpp [hunk 1]
```diff
@@ -14,14 +14,10 @@
...
 {
-MarchRobot::MarchRobot()
-{
-}
-
 MarchRobot::MarchRobot(::std::vector<Joint> jointList, ::std::string ifName, int ecatCycleTime)
 {
-  this->jointList = jointList;
-  ethercatMaster.reset(new EthercatMaster(&jointList, ifName, this->getMaxSlaveIndex(), ecatCycleTime));
+  this->jointList = std::move(jointList);
```
Comment (martijnvandermarel) [inline march_hardware/src/MarchRobot.cpp:19]: Is this still necessary?

File: march_hardware_interface/src/march_hardware_interface.cpp [hunk 2]
```diff
@@ -17,10 +17,9 @@ using joint_limits_interface::PositionJointSoftLimitsInterface;
 
 namespace march_hardware_interface
 {
-MarchHardwareInterface::MarchHardwareInterface(ros::NodeHandle& nh, AllowedRobot robotName) : nh_(nh)
+MarchHardwareInterface::MarchHardwareInterface(ros::NodeHandle& nh, AllowedRobot robotName)
+  : nh_(nh), marchRobot(HardwareBuilder(robotName).createMarchRobot())
```
Comment (martijnvandermarel) [inline march_hardware_interface/src/march_hardware_interface.cpp:21]: Nice!

File: march_hardware_builder/test/TestAllowedRobots.cpp [hunk 3]
```diff
@@ -15,11 +15,15 @@ class AllowedRobotTest : public ::testing::Test
 protected:
 };
 
-TEST_F(AllowedRobotTest, TestMarch3)
+TEST_F(AllowedRobotTest, TestMarch3Creation)
 {
   HardwareBuilder hardwareBuilder = HardwareBuilder(AllowedRobot::march3);
-  march4cpp::MarchRobot march3;
-  ASSERT_NO_THROW(march3 = hardwareBuilder.createMarchRobot());
+  ASSERT_NO_THROW(march4cpp::MarchRobot march3 = hardwareBuilder.createMarchRobot());
```
Comment (martijnvandermarel) [inline march_hardware_builder/test/TestAllowedRobots.cpp:21]: Change so it doesn't fail because of lack of empty constructor

**Review by martijnvandermarel — COMMENTED** [review]
I will approve once the test is fixed (edit: seems to be fixed now)

**Review by martijnvandermarel — APPROVED** [review]
Good fix

File: march_hardware/src/MarchRobot.cpp (hunk 1 above)
Comment (Ishadijcks) [reply march_hardware/src/MarchRobot.cpp:19]: Necessary no, but since the old variable is no longer needed we might as well move it instead of copying.

**Review by sopje — APPROVED** [review]
March_hardware_builder test passed

**Pull request merged**